src/Parsers/FunctionSecretArgumentsFinder.h

```diff
@@ -287,6 +317,48 @@ class FunctionSecretArgumentsFinder
             markSecretArgument(url_arg_idx + 4);
     }
 
+    std::string findIcebergStorageType()
```

Comment:
Why do you have `findIcebergStorageType` and `extractDynamicStorageType`?

Reply:
Couldn't you implement the same remove argument approach instead of doing the "skipIndices" thing?

Reply:
`findIcebergStorageType` is used in logic to mask secrets, different storage types has secrets on different position, so need to detect type to choose proper method (`findS3TableEngineSecretArguments` or `findAzureTableEngineSecretArguments`). I use "skip" thing because can't remove argument - this code is used to writing in logs, and add arguments must be there.
In my opinion here hard to reuse code, and other old methods like `findS3TableEngineSecretArguments` don't reuse code, have a logic duplicate instead. Code reusing makes code more complex here.